### PR TITLE
3 node cluster, and fix e2e-tests

### LINKED ISSUE #128: Local Kind Cluster Improvements
This is more of an enhancement request than a feature request related to the local kind cluster, and is just a suggestion that I think could improve the local experience and make it feel more realistic but I want to hear your thoughts.

When MSO is deployed, an alertmanager StatefulSet is created with 2 replicas.
```
k get po -n tenant-2
NAME                      READY   STATUS    RESTARTS   AGE
alertmanager-tenant-2-0   2/2     Running   0          4m16s
alertmanager-tenant-2-1   0/2     Pending   0          4m16s
```

The StatefulSet has podAntiAffinity to ensure the alertmanager pods go to different nodes.
```
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - podAffinityTerm:
              labelSelector:
                matchLabels:
                  app.kubernetes.io/component: alertmanager
                  app.kubernetes.io/part-of: tenant-2
              topologyKey: topology.kubernetes.io/zone
            weight: 100
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchLabels:
                app.kubernetes.io/component: alertmanager
                app.kubernetes.io/part-of: tenant-2
            topologyKey: kubernetes.io/hostname
```

**The problem**: Is that locally you will never have 2 nodes using the current kind configuration.
```
k get no 
NAME                 STATUS   ROLES                  AGE   VERSION
kind-control-plane   Ready    control-plane,master   16m   v1.22.4
```

To be able to fully see both replicas of the StatefulSet in action, you need 3 nodes (if you taint your control-plane).
- control-plane
- worker
- worker 

The `README.md` is very straight forward and has a very low barrier to entry, and I would want to keep it that way by still being able to spin up kind by copying and pasting just one block of code. Overvall, i believe  this would be a very useful part of the `README.md`, the e2e tests will still pass, and you have a environment that is ready for more advanced testing.

### PR DESCRIPTION
**Pull request by cmwylie19**
Closes #128

There are certain things that you cannot test on a single node. There are also parts of the application that do not work on a single node (The AlertManager StatefulSet with PodAntiAffinity) 

see #128 for more details.

This feature request is to give the use an alternative kind cluster configuration with the same low barrier to entry. (just copy and paste)

This setup gives the user 3 nodes, for a more production feel than a single node.
```
k get no 
NAME                 STATUS   ROLES                  AGE   VERSION
kind-control-plane   Ready    control-plane,master   50s   v1.22.4
kind-worker          Ready    <none>                 15s   v1.22.4
kind-worker2         Ready    <none>                 15s   v1.22.4
```

The end to end tests **do not break** when you run 3 nodes on the kind cluster. 

Now, you can get all alertmanager pods up from the `sts` and even look at the nodes they are sceduled on:
```
k get po -n tenant-2 -o wide
NAME                      READY   STATUS    RESTARTS   AGE   IP            NODE           NOMINATED NODE   READINESS GATES
alertmanager-tenant-2-0   2/2     Running   0          39s   10.244.1.9    kind-worker    <none>           <none>
alertmanager-tenant-2-1   2/2     Running   0          39s   10.244.2.8    kind-worker2   <none>           <none>
prometheus-tenant-2-0     3/3     Running   0          39s   10.244.1.10   kind-worker    <none>           <none>
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 bundle.Dockerfile
 RELEASE_BODY.md
 jsonnet/vendor
+config.yaml
```
Comment (sthaha) [inline .gitignore:6]: why is this required ? 🤔

File: README.md [hunk 2]
```diff
@@ -69,7 +69,21 @@ to the controller-gen CLI page in the [kubebuilder documentation](https://book.k
   ```sh
   kind create cluster --image kindest/node:v1.22.4
   ```
-
+  &nbsp; For more advanced tests, a 3 node cluster may be more suitable.
```
Comment (sthaha) [inline README.md:72]: How about we add this change to hack/kind/config.yaml since it is beneficial to have 3 node clusters to test HA.

File: test/e2e/framework/assertions.go [hunk 3]
```diff
@@ -191,7 +191,7 @@ func (f *Framework) GetOperatorPod(t *testing.T) *v1.Pod {
 
 	// get the operator deployment
 	operator := appsv1.Deployment{}
-	f.AssertResourceEventuallyExists("monitoring-stack-operator", "operators", &operator)(t)
+	f.AssertResourceEventuallyExists("monitoring-stack-operator-prometheus-operator", "operators", &operator)(t)
```
Comment (sthaha) [inline test/e2e/framework/assertions.go:194]: I am not sure why this is needed 🤔 but lets see if CI agrees :)

File: .gitignore (hunk 1 above)
Comment (cmwylie19) [reply .gitignore:6]: The idea behind this is to keep the config.yaml out of the base of the code if you were to have run a 3 node kind cluster. Looking at your last comment, this will not be necessary if we move this config.yaml to `hack/kind/config.yaml` 

```
cat <<EOF > config.yaml
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  image: kindest/node:v1.22.4
- role: worker
  image: kindest/node:v1.22.4
- role: worker
  image: kindest/node:v1.22.4
EOF
kind create cluster --config=config.yaml
```

File: test/e2e/framework/assertions.go (hunk 3 above)
Comment (cmwylie19) [reply test/e2e/framework/assertions.go:194]: This test fails already if I am not mistaken. The _only_ way I could get the tests to pass is by changing that. I just checked again on a clean repository

File: README.md (hunk 2 above)
Comment (cmwylie19) [reply README.md:72]: sounds good, done
